dev/io.openliberty.microprofile.config.internal.common/src/io/openliberty/microprofile/config/internal/common/ConfigIntrospector.java

```diff
@@ -0,0 +1,111 @@
+ */
+@Component(configurationPolicy = ConfigurationPolicy.IGNORE)
+public class ConfigIntrospector extends SensitiveIntrospector implements Introspector {
+
+    @Reference
+    private ConfigIntrospectionProvider introspectionProvider;
+
+    /**
+     * We have no metadata to indicate whether a given config value is sensitive, so this is a basic heuristic to avoid things which are likely to be sensitive, in addition to the
+     * protection offered by SensitiveIntrospector.
+     */
+    private final static Pattern MAYBE_SENSITIVE = Pattern.compile("password|pass$|pwd|key$", Pattern.CASE_INSENSITIVE);
+
+    @Override
+    public String getIntrospectorName() {
+        return "MicroProfileConfig";
+    }
+
+    @Override
+    public String getIntrospectorDescription() {
+        return "Lists the config properties available to each application via MicroProfile Config";
+    }
+
+    @Override
+    public void introspect(PrintWriter pw) {
+        Map<String, Set<Config>> configs = introspectionProvider.getConfigsByApplication();
+        for (Entry<String, Set<Config>> entry : configs.entrySet()) {
+            introspectApp(pw, entry.getKey(), entry.getValue());
+        }
+    }
+
+    private void introspectApp(PrintWriter pw, String appName, Set<Config> configs) {
+        pw.println();
+        pw.println("########################");
+        pw.println("Config for " + appName);
+        pw.println("########################");
+
+        for (Config config : configs) {
+            introspectConfig(pw, config);
+        }
+    }
+
+    private void introspectConfig(PrintWriter pw, Config config) {
+        try {
+            pw.println("Config: " + config.toString());
+            for (ConfigSource source : config.getConfigSources()) {
+                introspectConfigSource(pw, source);
+            }
+        } catch (Exception e) {
+            pw.printf("Exception introspecting config: %s", e);
+        }
+    }
+
+    private void introspectConfigSource(PrintWriter pw, ConfigSource source) {
+        try {
+            pw.printf(" - %s config source (ordinal %d)\n", source.getName(), source.getOrdinal());
+
+            Map<String, String> properties = source.getProperties();
+            if (properties == null) {
+                pw.println("     Cannot list properties from this config source because ConfigSource.getProperties() returned null.");
+                return;
+            }
+
+            for (Entry<String, String> entry : properties.entrySet()) {
+                String obscuredValue = getObscuredValue(entry.getKey(), entry.getValue());
+                pw.printf("   - %s = %s\n", entry.getKey(), obscuredValue);
+            }
+        } catch (Exception e) {
+            pw.printf("Exception introspecting config source: %s\n", e);
+        }
+    }
+
+    @Override
+    protected String getObscuredValue(String key, Object value) {
+        if (MAYBE_SENSITIVE.matcher(key).find()) {
```

Comment:
Can you add a test for this? A few variations of key name that should match the pattern?

Reply:
Good point, added